src/hooks/useHassuTable.tsx

```diff
@@ -11,6 +11,31 @@ import {
 } from "react-table";
 import { HassuTableProps } from "@components/HassuTable";
 import { Stack } from "@mui/material";
+import { Column, TableState } from "react-table";
+
+interface Test<D extends object> {
+  columns: readonly Column<D>[];
+  data: readonly D[];
+  initialState: Partial<TableState<D>> | undefined;
+}
+
+export const useHassuTableStabilizer = <D extends object>(tableOptions: Test<D>): HassuTableProps<D> => {
+  const [tableProps, setTableProps] = useState<HassuTableProps<D> | null>(null);
+
+  const newTableProps = useHassuTable<D>({
+    tableOptions,
+    useRowSelect: true,
+  });
+
+  useEffect(() => {
+    if (!tableProps) {
+      setTableProps(newTableProps);
+    }
+  }, [tableProps, setTableProps, newTableProps]);
+
+  if (!tableProps) return newTableProps;
+  return tableProps;
+};
```

Comment:
Tonne on tainnut jäädä testailulta tyypille nimeks Test.

Toiseks tää stabilizer nyt raa'asti vaan estää tablePropsien päivittyksen vaikka tälle hookille annetut propsit muuttusikin. Mun käsityksen mukaa useHassuTable -hookkiin jos lisäis memoizausta muutamaan kohtaan niin sais vähennettyä renderaus kertoja kuitenkaan estämättä sen uudelleen renderausta. Testailin lisäämällä memoizointia tableHooks, defaultTableOptions, tableProps ja returnattavaan arvoon. Näin sain ne moneen kertaan renderoinnit estettyä.

Reply:
Muuten mun mielestä OK 👍

Reply:
Keskusteltiin, että teen oman PR:n näistä useHassuTable-muutoksista ja riisutaan ne tästä PR:stä kokonaan pois